### DISCUSSION TITLE
IndexError:arrays used as indices must be of integer (or boolean) type

### DISCUSSION THREAD

**Discussion by josh-001** [roboflow/supervision#900] [Q&A]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Question

while running yolo-nas and sam for segmentation of my sample video i am getting error as
ERROR: 
detections = detections[detections.area == np.max(detections.area)]
---> 62   segmented_mask = mask_annotator.annotate(scene=frame, detections=detections)
     63   return segmented_mask

[/usr/local/lib/python3.10/dist-packages/supervision/annotators/core.py](https://localhost:8080/#) in annotate(self, scene, detections, custom_color_lookup)
    250             )
    251             mask = detections.mask[detection_idx]
--> 252             colored_mask[mask] = color.as_bgr()
    253 
    254         scene = cv2.addWeighted(colored_mask, self.opacity, scene, 1 - self.opacity, 0)

IndexError: arrays used as indices must be of integer (or boolean) type

**Comment by onuralpszr** [comment]
@josh-001  could you share full collab demo please

**Comment by josh-001** [comment]
https://colab.research.google.com/drive/1GSEf1ua-MklrxC8z8KpRkzLpYRFIDDT1?usp=sharing

this is my colab link.

**Comment by SkalskiP** [comment]
Hi @josh-001 👋🏻 ! It does not look like a bug related to `supervision`. Looks more like something related to SAM result processing. Wile reading your code I noticed, they you do not use `sv.Detections.from_sam(sam_result=sam_result)`. Here is a [colab](https://colab.research.google.com/github/roboflow-ai/notebooks/blob/main/notebooks/how-to-segment-anything-with-sam.ipynb#scrollTo=bdT3xT2AkS4g) where we show how to combo SAM and Supervision.

**Comment by SkalskiP** [comment]
I'll also convert this issue into a discussion and put it in the Q&A section.
